CRM/Financial/BAO/FinancialItem.php

```diff
@@ -286,25 +286,25 @@ public static function checkContactPresent($contactIds, &$error) {
   }
 
   /**
-   * Get last financial item data.
+   * Get most relevant previous financial item relating to the line item.
    *
-   * @param int $entityId
+   * This function specifically excludes sales tax.
    *
-   * @param string $entityTable
+   * @param int $entityId
    *
    * @return object CRM_Core_DAO
    */
-  public static function getPreviousFinancialItem($entityId, $entityTable = 'civicrm_line_item') {
-    $queryParams = array(
-      1 => array($entityId, 'Integer'),
-      2 => array($entityTable, 'String'),
+  public static function getPreviousFinancialItem($entityId) {
+    $params = array(
+      'entity_id' => $entityId,
+      'entity_table' => 'civicrm_line_item',
+      'options' => array('limit' => 1, 'sort' => 'id DESC'),
     );
-    $query = 'SELECT id, description, status_id, financial_account_id 
-      FROM civicrm_financial_item
-      WHERE entity_id = %1 AND entity_table = %2 ORDER BY id DESC LIMIT 1';
-    $prevFinancialItem = CRM_Core_DAO::executeQuery($query, $queryParams);
-    $prevFinancialItem->fetch();
-    return $prevFinancialItem;
+    $salesTaxFinancialAccounts = civicrm_api3('FinancialAccount', 'get', array('is_tax' => 1));
```

Comment:
Sorry, I don't agree with is_tax = 1 criteria, since is_tax = 1 is set for Financial Account having financial_account_type as Liability. User may configure Financial Type having relationship as 'Deferred Revenue Is' with Financial Account for is_tax =1 which will end up Fatal error when trying to update Deferred Revenue contribution.